src/WhereConditions/WhereConditionsBaseDirective.php

```diff
@@ -178,18 +176,6 @@ protected static function assertValidColumnReference(string $column): void
         }
     }
 
-    protected function nestedRelatedModel(Model $model, string $nestedRelationPath): Model
```

Comment:
Pretty sure that change is just straight up removing a feature. Looks like it was not tested, but I can see the use case behind it.

Reply:
I'll try to create a test for a nested relationship and let you know. I think that could be the potential break

Reply:
I tried to create the test `testWhereConditionsHasNestedTables` where I traverse multiple nested relationships and it still pass. I'm unsure if it is what you meant though. Checking `WhereConditionsDirectiveTest` class a can see many tests similar to this one, with the only difference in this case the same table is used multiple times.

My understanding is that the logic from `nestedRelatedModel` is already implemented in `whereHas`/`has` Builder method, but I could be completely wrong

Reply:
The test you wrote is not really representative of the potential issue, since no matter how deep you nest `parent.parent....`, it always stays within the same model.

I created https://github.com/nuwave/lighthouse/pull/1807/files, but it seems there are some problems with it. Looks like the implementation is already bugged.

Reply:
Ah I understand now. It's not using the inner most relation table name. I'll check if I'm able to fix this

Reply:
I just found out the problem was related to `posts` relationship in `Category` model. Related class was set to `Category` instead of `Post`.

Changing that both the hold implementation and the new one work as expected (at least the test case you created passes)

Reply:
Great find, thank you.